pkg/standalone-utils/test/foundry/BalancerFeeBurner.t.sol

```diff
@@ -0,0 +1,259 @@
+            ),
+            address(feeSweeper)
+        );
+
+        // Allow the fee sweeper to burn.
+        authorizer.grantRole(feeBurnerAuth.getActionId(IProtocolFeeBurner.burn.selector), address(feeSweeper));
+        authorizer.grantRole(feeBurnerAuth.getActionId(IProtocolFeeBurner.burn.selector), alice);
+
+        authorizer.grantRole(feeBurnerAuth.getActionId(IBalancerFeeBurner.setBurnPath.selector), alice);
+
+        vm.prank(admin);
+        feeSweeper.addProtocolFeeBurner(feeBurner);
+
+        // Craate dai -> weth -> usdc path
+        (daiWethPool, ) = _createPool([address(dai), address(weth)].toMemoryArray(), "pool");
+        (wethUsdcPool, ) = _createPool([address(weth), address(usdc)].toMemoryArray(), "pool");
+
+        approveForPool(IERC20(daiWethPool));
+        approveForPool(IERC20(wethUsdcPool));
+
+        vm.startPrank(lp);
+        _initPool(daiWethPool, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        _initPool(wethUsdcPool, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        vm.stopPrank();
+    }
+
+    function testSweepAndBurn() public {
+        IBalancerFeeBurner.SwapPathStep[] memory steps = new IBalancerFeeBurner.SwapPathStep[](1);
+        steps[0] = IBalancerFeeBurner.SwapPathStep({ pool: pool, tokenOut: usdc });
+
+        vm.prank(alice);
+        feeBurner.setBurnPath(dai, steps);
+
+        // Set up the sweeper to be able to burn.
+        vm.startPrank(admin);
+        feeSweeper.setTargetToken(usdc);
+
+        // Put some fees in the Vault.
+        vault.manualSetAggregateSwapFeeAmount(pool, dai, DEFAULT_AMOUNT);
+
+        vm.expectEmit();
+        emit IProtocolFeeBurner.ProtocolFeeBurned(pool, dai, DEFAULT_AMOUNT, usdc, DEFAULT_AMOUNT, feeRecipient);
+
+        feeSweeper.sweepProtocolFeesForToken(pool, dai, DEFAULT_AMOUNT, orderDeadline, feeBurner);
+        vm.stopPrank();
+    }
+
+    function testBurnWithOneHop() external {
+        IBalancerFeeBurner.SwapPathStep[] memory steps = new IBalancerFeeBurner.SwapPathStep[](1);
+        steps[0] = IBalancerFeeBurner.SwapPathStep({ pool: pool, tokenOut: usdc });
+
+        uint256 daiBalanceBefore = dai.balanceOf(alice);
+        uint256 usdcBalanceBefore = usdc.balanceOf(alice);
+        uint256 daiVaultBalanceBefore = dai.balanceOf(address(vault));
+        uint256 usdcVaultBalanceBefore = usdc.balanceOf(address(vault));
+
+        vm.startPrank(alice);
+        feeBurner.setBurnPath(dai, steps);
+
+        IERC20(address(dai)).forceApprove(address(feeBurner), TEST_BURN_AMOUNT);
+        feeBurner.burn(address(0), dai, TEST_BURN_AMOUNT, usdc, MIN_TARGET_TOKEN_AMOUNT, alice, orderDeadline);
+        vm.stopPrank();
+
+        uint256 daiBalanceAfter = dai.balanceOf(alice);
+        uint256 usdcBalanceAfter = usdc.balanceOf(alice);
+        uint256 daiVaultBalanceAfter = dai.balanceOf(address(vault));
+        uint256 usdcVaultBalanceAfter = usdc.balanceOf(address(vault));
```

Comment:
This is fine, although you can also use the helper in the base layer `getBalances` for alice before and after, which should give you all of these nicely packed into a struct.